tests/test_parts_manager_split_tags.py

```diff
@@ -0,0 +1,573 @@
+    assert len(events) == 1
+    assert events[0] == snapshot(
+        PartDeltaEvent(
+            index=0, delta=TextPartDelta(content_delta='nk>', part_delta_kind='text'), event_kind='part_delta'
+        )
+    )
+    assert manager.get_parts() == snapshot([TextPart(content='<think>', part_kind='text')])
+
+
+def test_handle_text_deltas_false_start_then_real_tag():
+    """Test buffering a false start, then processing real content."""
+    manager = ModelResponsePartsManager()
+    thinking_tags = ('<think>', '</think>')
+
+    # Chunk 1: "<th" - could be tag start, buffer it
+    events = list(manager.handle_text_delta(vendor_part_id='content', content='<th', thinking_tags=thinking_tags))
+    assert len(events) == 0  # Buffered
+
+    # Chunk 2: "is is text" - proves it's not a tag, flush buffer
+    events = list(
+        manager.handle_text_delta(vendor_part_id='content', content='is is text', thinking_tags=thinking_tags)
+    )
+    assert len(events) == 1
+    assert events[0] == snapshot(
+        PartStartEvent(index=0, part=TextPart(content='<this is text', part_kind='text'), event_kind='part_start')
+    )
+    assert manager.get_parts() == snapshot([TextPart(content='<this is text', part_kind='text')])
+
+
+def test_buffered_content_exceeds_tag_length():
+    """Test that buffered content longer than tag is flushed (covers line 231)."""
+    manager = ModelResponsePartsManager()
+    thinking_tags = ('<think>', '</think>')
+
+    # To hit line 231, we need:
+    # 1. Buffer some content
+    # 2. Next chunk starts with '<' (to pass first check)
+    # 3. Combined length >= tag length
+
+    # First chunk: exactly 6 chars
+    events = list(manager.handle_text_delta(vendor_part_id='content', content='<think', thinking_tags=thinking_tags))
+    assert len(events) == 0  # Buffered
+
+    # Second chunk: starts with '<' so it checks _could_be_tag_start
+    # Combined will be '<think<' (7 chars) which equals tag length '<think>' (7 chars)
+    events = list(manager.handle_text_delta(vendor_part_id='content', content='<', thinking_tags=thinking_tags))
+    # 7 >= 7 is True, so line 231 returns False
+    assert len(events) == 1
+    assert events[0] == snapshot(
+        PartStartEvent(index=0, part=TextPart(content='<think<', part_kind='text'), event_kind='part_start')
+    )
+    assert manager.get_parts() == snapshot([TextPart(content='<think<', part_kind='text')])
+
+
+def test_complete_thinking_tag_no_vendor_id():
+    """Test complete thinking tag with vendor_part_id=None (covers lines 161-164)."""
+    manager = ModelResponsePartsManager()
+    thinking_tags = ('<think>', '</think>')
+
+    # Complete start tag with vendor_part_id=None goes through simple path
+    # This covers lines 161-164 in _handle_text_delta_simple
+    events = list(manager.handle_text_delta(vendor_part_id=None, content='<think>', thinking_tags=thinking_tags))
```

Comment:
If the model only ever outputs `<think>`, without any text following it, I think that should be a `<think>` text part